types/rlp_unmarshal.go

```diff
@@ -60,12 +80,26 @@ func (b *Block) UnmarshalRLPFrom(p *fastrlp.Parser, v *fastrlp.Value) error {
 		return err
 	}
 
-	for _, txn := range txns {
-		bTxn := &Transaction{}
-		if err := bTxn.UnmarshalRLPFrom(p, txn); err != nil {
+	for i := 0; i < len(txns); i++ {
+		txType := LegacyTx
+		if txns[i].Type() == fastrlp.TypeBytes {
+			if err = txType.UnmarshalRLPFrom(p, txns[i]); err != nil {
+				return err
+			}
+
+			i++
+		}
+
+		bTxn := &Transaction{
+			Type: txType,
+		}
+
+		if err = bTxn.UnmarshalRLPFrom(p, txns[i]); err != nil {
 			return err
 		}
 
+		bTxn.ComputeHash()
```

Comment:
```suggestion
		_ := bTxn.ComputeHash()
```

or prefer to obtain the result and work with it explicitly

Reply:
But it just returns the same object. Do you think we really need it?

Reply:
Then why it modifies the object inside the function and returns the same object? It's a function with hidden side effects, a code smell, https://weblogs.asp.net/podwysocki/side-effecting-functions-are-code-smells, preferably return the new object, I'd refactor that function if possible.

Reply:
Done